packages/devtools_app/test/test_infra/flutter_test_driver.dart

```diff
@@ -414,12 +414,12 @@ class FlutterRunTestDriver extends FlutterTestDriver {
   Future<int> detach() async {
     if (vmService != null) {
       debugPrint('Closing VM service');
-      await vmService.dispose();
+      await vmService!.dispose();
```

Comment:
Assign `vmService` to a local so we only do one null check.

Reply:
If there were at least two null assersions, it would make sense. There is just one.

Reply:
If you do this:

```dart
final service = vmService;
if (service != null) {
  await service.dispose();
}
```

We only perform one null check instead of two. Anywhere with the pattern:

```dart
if (foo != null) {
  foo!.doSomething();
}
```
should either be updated to:

```
final localFoo = foo;
if (localFoo != null) {
  localFoo.doSomething();
}
```

or

```
foo?.doSomething();
```

Reply:
Hm. What's the point to update 

```dart
if (foo != null) {
  foo!.doSomething();
}
```
with
```dart
final localFoo = foo;
if (localFoo != null) {
  localFoo.doSomething();
}
```
?

The second piece seems less readable to me. Is there performance improvement?

Reply:
Yes, we save on null-check. On other side, the additional variable declaration adds to both performance and cognitive load.
There are two options how we can resolve it: (1) agree it is subjective and should be up to author or (2) declare a team standard for such cases. What do you think?

Reply:
> the additional variable declaration adds to both performance and cognitive load.

The additional variable should be GC'd once it is out of scope.

Let's declare a team standard to avoid multiple null assertions when possible. Consistency across the codebase will help with cognitive load over time.

Reply:
Sounds good.

Is it written standard or de-facto standard throughout the dart core libraries? I do not remember reading it. 
As it is not obvious, and there are pros and cons in both approaches, should we write it to make contribution easier for new members and external contributors? 

In one of these places, for example:
https://dart.dev/guides/language/effective-dart/usage#null
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

Reply:
@leafpetersen @munificent do you have an opinion on whether we should recommend this style to users?

Reply:
@munificent may have more to say here.  Obviously, when applicable the `foo?.bar()` approach is fine (I like it).  The style guide already softly [recommends](https://dart.dev/guides/language/effective-dart/usage#consider-copying-a-nullable-field-to-a-local-variable-to-enable-type-promotion) the local variable approach ("Consider").  I personally don't mind using a "!" when there's only one occurrence of the variable (even though there is the redundant check), but I'm pretty sure @lrhn yelled at me the last time I did that in a CL, so.... :)

Reply:
Thank you for the link. I see there are three usages of '!' in the example. And, yes, the question is what we think about single usage.

There are actually two questions:
1. Do we want to recommend local variable for this case to Dart users?
2. Do we want to document the style as a standard across Dart code libraries?

Reply:
TBH, my preferred answer to this is that you should use the new pattern feature that @munificent is working on to do the null check and the local variable binding in one step inside of the condition.  Unfortunately he hasn't shipped it yet, dunno what the hold up is... :). 

So I'll defer to others as to what we want to do in the interim, but I would caution against putting large effort into standardizing/cleaning up around one of the existing solutions given that I expect (hope) that we will be shipping a better solution within a year.

Reply:
That's cool! Let's wait then. I am curious how the code would look like.